pkg/controller/drain.go

```diff
@@ -52,15 +51,17 @@ type DrainOptions struct {
 	Force              bool
 	GracePeriodSeconds int
 	IgnoreDaemonsets   bool
-	Timeout            time.Duration
-	PvDetachTimeout    time.Duration
-	DeleteLocalData    bool
-	nodeName           string
-	Out                io.Writer
-	ErrOut             io.Writer
-	Driver             driver.Driver
-	pvcLister          corelisters.PersistentVolumeClaimLister
-	pvLister           corelisters.PersistentVolumeLister
+	// Timeout has been deprecated and has no effect.
+	Timeout         time.Duration
```

Comment:
I am somehow skeptical about not keeping the global-timeout. @ggaurav10 raised a nice point, while offline discussion.

Basically, an uninformed user can halt the rollout or make it really long for certain machines, by keeping extremely long gracefulTimePeriod on Pods. I was hoping more of a solution, which gives lesser control to the user, especially in terms of managing machines.

I can remember having certain counter-arguments over this concern, can you please delineate in case you are aware of any.

Reply:
@hardikdr @ggaurav10 I share the concerns about not having a defence against misconfigured long `terminationGracePeriodSeconds`. But not honouring `terminationGracePeriodSeconds` is also not an option. Otherwise we would be opting not to support some genuine use-cases (some heavy stateful workload).

So, the main question .  is what is the best way to defend. I can think of several ways including a global timeout. Please note: these need not be mutually exclusive.
1. Global timeout
What value would be good? `1/2/6/12/24h`? 
I am a bit reluctant about this because of the side effects on the complications on the code as well as undesirable choices it creates. Also, this goes against fail-fast principle.
1. Validate and enforce upper limit on `terminationGracePeriods` via validating webhooks or some other similar lateral mechanism (`PodPresets`?)
Fails fast. It catches the problem early when the pod is deployed and not when it is evicted and then it leaves a mess like above. For flexibility, the upper limit for `terminationGracePeriods` could be configurable in the `Shoot` resource. It basically acts like a declarative policy. I prefer this the most.
1. Expose granular metrics about`terminationGracePeriodSeconds`, actual termination duration, volume detach duration, successful pod eviction/volume detach within timeout, failed pod eviction/volume detach within timeout etc. at the granularity of both the pods and the node. This can act as diagnostics both for alerts as well as help devops in identifying the problem during support.

There could be more. WDYT?

Reply:
>Global timeout
What value would be good? 1/2/6/12/24h?
I am a bit reluctant about this because of the side effects on the complications on the code as well as undesirable choices it creates. Also, this goes against fail-fast principle.

In case the global timeout (1/2/6/12/24h) is less than the timeout value calculated using `terminationGraecPeriodSeconds`, we would still `fail fast`er.

Although I agree that 2nd option adheres to `fail fast` principle better, however, the reluctance of having complications in the code or making undesirable choices would still exist; that would just be moved to another component. WDYT?

Reply:
@ggaurav10 Yes. But by fail fast, I meant when the pod is created (fastest 😉)  rather than when the pod is about to be evicted. But it would be *faster*, yes 😄